scripts/layouts_tool/layoutstoolmain.cpp

```diff
@@ -0,0 +1,335 @@
+        utils::qout() << (loaded ? "(5): Quit" : "(3): Quit") << endl;
+
+        // Prompt user for choice
+        utils::qin() >> menuChoice;
+
+        switch(menuChoice) {
+            case 1: {
+                utils::clearTerminal();
+                utils::qout() << "Please tell me the path to the layouts cpp file (no spaces please): " << endl;
+                utils::qin() >> m_FilePath;
+                m_pLayoutsFileHandler->open(m_FilePath, m_Layouts);
+                break;
+            }
+
+            case 2: {
+                utils::qout() << "Save file..." << endl;
+                QFile f(m_FilePath);
+                m_pLayoutsFileHandler->save(f, m_Layouts);
+                break;
+            }
+
+            case 3: {
+                if (loaded) {
+                    editLayoutMenu();
+                } else {
+                    userWantsToQuit = true;
+                    quit();
+                }
+                break;
+            }
+
+            case 4: {
+                if (loaded) {
+                    // Miscellaneous tools
+                    miscToolsMenu();
+                    break;
+                }
+            }
+
+            case 5: {
+                if (loaded) {
+                    userWantsToQuit = true;
+                    quit();
+                    break;
+                }
+            }
+
+            default:
+                qDebug() << "ERROR! You have selected an invalid choice.";
+                break;
+        }
+    } while (!userWantsToQuit);
+}
+
+void LayoutsToolMain::editLayoutMenu() {
+    bool loaded = !m_FilePath.isEmpty();
+    if (!loaded) {
+        qDebug() << "Can't edit any layout, any layout loaded.";
+        return;
+    }
+
+    bool backToMain = false;
+    do {
+        int menuChoice = 0;
+
+        // Print menu
+        utils::clearTerminal();
+        utils::qout() << "********** LAYOUTS TOOL - EDIT **********" << endl;
+        utils::qout() << "Editing file: " << m_FilePath << endl;
+        utils::qout() << "(1): Remove layout" << endl;
+        utils::qout() << "(2): Add layout" << endl;
+        utils::qout() << "(3): Back to main menu" << endl;
+
+        // Prompt user for choice
+        utils::qin() >> menuChoice;
+
+        switch(menuChoice) {
+            case 1: {
+                // Remove layouts
+                removeLayoutMenu();
+                break;
+            }
+
+            case 2: {
+                // Add layout
+                addLayoutMenu();
+                break;
+            }
+
+            case 3: {
+                // Back to main menu
+                backToMain = true;
+                break;
+            }
+
+            default:
+                qDebug() << "ERROR! You have selected an invalid choice.";
+                break;
+        }
+    } while (!backToMain);
+}
+
+void LayoutsToolMain::addLayoutMenu() {
+    QString kbdLocale = utils::inputLocaleName();
```

Comment:
This somehow ambiguous because different layouts use the same QLocale code: 
See: http://cep.xray.aps.anl.gov/software/qt4-x11-4.2.2-browser/dd/d50/qkeymapper__x11__p_8cpp-source.html
This will be not an issue if we hit the most common keyboard used for the selected language, but if we fail? The users will tell us :-) 

It is not a real issue, because the user can make its own mapping for a non standard layout. But we should state this issue clear somewhere. How can the user check which keyborad layout we have used without hitting the any key? Could we give a list which layout we have used for generating the language dependent layouts? 
Something like we have found in the qkeymapper__x11__p_8cpp name field e.g "us:intl"